lib/stateTransition/validation/validateStateTransitionFeeFactory.js

```diff
@@ -35,13 +35,16 @@ function validateStateTransitionFeeFactory(
 
     switch (stateTransition.getType()) {
       case stateTransitionTypes.IDENTITY_CREATE: {
-        result.merge(await validateLockTransaction(stateTransition));
+        const validateLockTransactionResult = await validateLockTransaction(stateTransition);
+
+        if (!validateLockTransactionResult.isValid()) {
+          result.merge(validateLockTransactionResult);
 
-        if (!result.isValid()) {
           return result;
         }
 
-        const output = result.getData();
+        const output = validateLockTransactionResult.getData();
+
         balance = output.satoshis;
```

Comment:
1 satoshi is 1000 credits. I guess we need a function that converts satoshis to credits.